index.js

```diff
@@ -1,219 +1,260 @@
-                        this.focusField(Math.min(filledPinCount, pinCount - 1))
-                    }}
-                >
-                    <View
-                        style={{ flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', width: '100%', height: '100%' }}
-                    >
-                        {this.renderTextFields()}
-                    </View>
-                </TouchableWithoutFeedback>
-            </View>
-        );
-    }
-}
\ No newline at end of file
+	static propTypes = {
+		pinCount: PropTypes.number,
+		codeInputFieldStyle: PropTypes.object,
+		codeInputHighlightStyle: PropTypes.object,
+		onCodeFilled: PropTypes.func,
+		onCodeChanged: PropTypes.func,
+		autoFocusOnLoad: PropTypes.bool,
+		code: PropTypes.string
+	};
+
+	static defaultProps = {
+		pinCount: 6,
+		codeInputFieldStyle: null,
+		codeInputHighlightStyle: null,
+		onCodeFilled: null,
+		autoFocusOnLoad: true
+	};
+
+	fields = [];
+
+	constructor(props) {
+		super(props);
+		const { code } = props;
+		this.state = {
+			digits: code === undefined ? [] : code.split(''),
+			selectedIndex: 0
+		};
+	}
+
+	componentDidMount() {
+		this.copyCodeFromClipBoardOnAndroid();
+		this.bringUpKeyBoardIfNeeded();
+		this.keyboardDidHideListener = Keyboard.addListener(
+			'keyboardDidHide',
+			this.handleKeyboardDidHide
+		);
+	}
+
+	componentWillUnmount() {
+		if (this._timer) {
+			clearInterval(this._timer);
+		}
+		this.keyboardDidHideListener.remove();
+	}
+
+	copyCodeFromClipBoardOnAndroid = () => {
+		if (Platform.OS === 'android') {
+			this.checkPinCodeFromClipBoard();
+			this._timer = setInterval(() => {
+				this.checkPinCodeFromClipBoard();
+			}, 400);
+		}
+	};
+
+	bringUpKeyBoardIfNeeded = () => {
+		const { autoFocusOnLoad, pinCount } = this.props;
+		const digits = this.getDigits();
+		const focusIndex = digits.length ? digits.length - 1 : 0;
+		if (focusIndex < pinCount && autoFocusOnLoad) {
+			this.focusField(focusIndex);
+		}
+	};
+
+	getDigits = () => {
+		const { digits: innerDigits } = this.state;
+		const { code } = this.props;
+		return code === undefined ? innerDigits : code.split('');
+	};
+
+	handleKeyboardDidHide = () => {
+		this.blurAllFields();
+	};
+
+	notifyCodeChanged = () => {
+		const { digits } = this.state;
+		const code = digits.join('');
+		const { onCodeChanged } = this.props;
+		if (onCodeChanged) {
+			onCodeChanged(code);
+		}
+	};
+
+	checkPinCodeFromClipBoard = () => {
+		const { pinCount } = this.props;
+		const { digits } = this.state;
+		Clipboard.getString()
+			.then(code => {
+				if (
+					this.hasCheckedClipBoard &&
+					code.length === pinCount &&
+					this.clipBoardCode !== code
+				) {
+					this.setState(
+						{
+							digits: code.split('')
+						},
+						() => {
+							this.blurAllFields();
+							this.notifyCodeChanged();
+							digits.join('');
```

Comment:
Hi @fabiohfab, we didn't hear from you for a long time. We had went through the code manually and found the only two differences are `digits.join('')` and `const { digits } = this.state`. We found adding these two lines of code couldn't achieve the functionality that you suggested in the PR. Please feel free to talk with us if we misunderstood your intention. Glad to discuss further. Thanks!